src/core/linux/SDL_dbus.c

```diff
@@ -534,6 +534,62 @@ char *SDL_DBus_GetLocalMachineId(void)
 
     return NULL;
 }
+
+/*
+ * Convert file drops with mime type "application/vnd.portal.filetransfer" to file paths
+ * Result must be freed with dbus->free_string_array().
+ * https://flatpak.github.io/xdg-desktop-portal/#gdbus-method-org-freedesktop-portal-FileTransfer.RetrieveFiles
+ */
+char **SDL_DBus_GetPortalFilePath(char *key, int *file_path_count)
+{
+    DBusError err;
+    char** file_paths = NULL;
+    DBusMessageIter iter, iterDict;
+
+    DBusMessage *reply = NULL;
+    DBusMessage *msg = dbus.message_new_method_call("org.freedesktop.portal.Documents",    /* Node */
+                                                    "/org/freedesktop/portal/documents",   /* Path */
+                                                    "org.freedesktop.portal.FileTransfer", /* Interface */
+                                                    "RetrieveFiles");                      /* Method */
+
+    if (!dbus.session_conn) {
+        /* We either lost connection to the session bus or were not able to
+         * load the D-Bus library at all. */
+        return NULL;
+    }
+
+    dbus.error_init(&err);
+
+    dbus.message_append_args(msg, DBUS_TYPE_STRING, &key, DBUS_TYPE_INVALID);
+
+    dbus.message_iter_init_append(msg, &iter);
+    if (!dbus.message_iter_open_container(&iter, DBUS_TYPE_ARRAY, "{sv}", &iterDict) || !dbus.message_iter_close_container(&iter,  &iterDict)) {
```

Comment:
how could this fail?

Reply:
I was wondering this myself, I was just practising defensive programming.
The dbus code says its for out of memory: [dbus-message.c:2919](https://gitlab.freedesktop.org/dbus/dbus/-/blob/172aa1c474759fe96b2072edcc22622ed1cb0510/dbus/dbus-message.c?page=3#L2919)

Reply:
ok then it's worth adding a failure case for that